Adds AND or OR filtering on style

johnnyblasta opened the pull request:
More filtering on style as mentioned in #858

In thematic style add ' && ' to do AND filtering or ' || ' to do OR filtering.
Can be used multiple times, but not combine AND and OR filtering on the same filter

```
"filter": "[zon] == '5' && [status] == 'Avstängd'"
"filter": "[zon] == '5' || [status] == 'Avstängd'"
```

jokd commented:
It doesn't work for me, tried this on the origo-logo-style:
"filter": "[id] == '3' || [name] == 'Sundsvall'"

Only the latter shows up in the map

johnnyblasta commented:
Fixed a problem with OR logic.
Changed the syntax to use AND/OR instead of &&/||

```
"filter": "[zon] == '5' AND [status] == 'Avstängd'"
"filter": "[zon] == '5' OR [status] == 'Avstängd'"
```
